crates/uv/tests/tool_run.rs

```diff
@@ -823,6 +823,135 @@ fn tool_run_without_output() {
+        [project]
+        name = "foo"
+        version = "1.0.0"
+        requires-python = ">=3.8"
+        dependencies = ["anyio", "sniffio==1.3.1"]
+        "#
+    })?;
+
+    let test_script = context.temp_dir.child("main.py");
+    test_script.write_str(indoc! { r"
+        import sniffio
+       "
+    })?;
+
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("--with-editable")
+        .arg("./src/black_editable")
+        .arg("--with")
+        .arg("iniconfig")
+        .arg("flask")
+        .arg("--version")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.12.[X]
+    Flask 3.0.2
+    Werkzeug 3.0.1
+
+    ----- stderr -----
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + black==0.1.0 (from file://[TEMP_DIR]/src/black_editable)
+     + blinker==1.7.0
+     + click==8.1.7
+     + flask==3.0.2
+     + iniconfig==2.0.0
+     + itsdangerous==2.1.2
+     + jinja2==3.1.3
+     + markupsafe==2.1.5
+     + werkzeug==3.0.1
+    "###);
+
+    // Requesting an editable requirement should install it in a layer, even if it satisfied
+    uv_snapshot!(context.filters(), context.tool_run().arg("--with-editable").arg("./src/anyio_local").arg("flask").arg("--version"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.12.[X]
+    Flask 3.0.2
+    Werkzeug 3.0.1
+
+    ----- stderr -----
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + anyio==4.3.0+foo (from file://[TEMP_DIR]/src/anyio_local)
+     + blinker==1.7.0
+     + click==8.1.7
+     + flask==3.0.2
+     + itsdangerous==2.1.2
+     + jinja2==3.1.3
+     + markupsafe==2.1.5
+     + werkzeug==3.0.1
+    "###);
+
+    // Requesting the project itself should use a new environment.
+    uv_snapshot!(context.filters(), context.tool_run().arg("--with-editable").arg(".").arg("flask").arg("--version"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.12.[X]
+    Flask 3.0.2
+    Werkzeug 3.0.1
+
+    ----- stderr -----
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + anyio==4.3.0
+     + blinker==1.7.0
+     + click==8.1.7
+     + flask==3.0.2
+     + foo==1.0.0 (from file://[TEMP_DIR]/)
+     + idna==3.6
+     + itsdangerous==2.1.2
+     + jinja2==3.1.3
+     + markupsafe==2.1.5
+     + sniffio==1.3.1
+     + werkzeug==3.0.1
+    "###);
+
+    // If invalid, we should reference `--with`.
+    uv_snapshot!(context.filters(), context.tool_run().arg("--with").arg("./foo").arg("flask").arg("--version"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: Distribution not found at: file://[TEMP_DIR]/foo
+    "###);
```

Comment:
Just a quick comment, looks like this _doesn't_ reference `--with` like we'd hope :)

Reply:
Thank you for your review :)
I'll try working on it for a few more days 👍🏽

Reply:
I've implemented the feature similarly to the pull request, but I'm running into an issue where an `anyhow::Error` is being returned, which is preventing me from adding context to the error. I need to investigate this further, but if you have any insights or know of a possible cause, I would greatly appreciate your input.

Reply:
@zanieb 
Could you please check it once?

Reply:
73085aec6f48da64fb2edd63d7e3df38177466d2 has the relevant fix